### PR TITLE
Add support for Postgresql

### PR DESCRIPTION
**Pull request by leon-vv**
This pull request adds support for Postgresql (for a very limited number of data types). The database specific parts are located in the files ending with '-interface.js'. Adding other databases is therefore trivial.

Please check that I have not broken the Mysql interface.

### CODE REVIEW

**Comment by TehShrike** [comment]
Cool!

I'd rather not have to support the complexity of two database engines in this project - how do you feel about publishing the postgres fork as `joi-pgsql` or something similar, and I'll link to it in the readme of this project?

**Comment by leon-vv** [comment]
I think that would be a suboptimal solution. There is a good chance someone comes along and adds support for another database (SQLite for example). In that case splitting the code bases would cause a lot of duplicated work.

Furthermore, this PR does not make the code a lot more complex. The `information_schema.columns` table seems to be standarized, so the only major change is passing around the interface module (intf parameter).

I would be happy to do the support for the Postgresql code. Also, if there is anything you don't like about this change just let me know and I'll correct it.

File: build.js [hunk 1]
```diff
@@ -28,49 +30,60 @@ function unrollEnum(col) {
...
-			if (column.columnType.indexOf('unsigned') === -1) {
-				max = getSignedValue(max)
-				min = -1 * (max + 1)
+			checks += '.number().integer()'
+            
+			if (!intf.hasUnsigned) { 
+			    let [min, max] = bytesToSignedRange(bytes);
+                checks += '.max(' + max + ').min(' + min + ')'
 			}
+            else if(column.columnType.indexOf('unsigned') !== -1) {
```
Comment (TehShrike) [inline build.js:46]: It looks like both of these `if` blocks contain the same code, so this should probably be `if (!intf.hasUnsigned || column.columnType.indexOf('unsigned') !== -1)`

File: build.js [hunk 2]
```diff
@@ -28,49 +30,60 @@ function unrollEnum(col) {
...
+    
 		var checks = ''
-		if (maxIntValues[column.dataType]) {
-			checks += '.number().integer()'
+        var bytes = intf.intToBytes[column.dataType];
+
+		if (bytes) {
 
-			var min = 0
-			var max = maxIntValues[column.dataType]
```
Comment (TehShrike) [inline build.js:37]: It looks like after this change, there is no min/max checking when the db engine supports unsigned numbers, and the column is signed.

**Comment by TehShrike** [comment]
A SQLite version would be sweet.  If someone forked this library and published joi-sqlite, that would be awesome.

If there would be any significant code shared between the libraries, I would be happy to move that to its own module.  Moving the code in `build.js` to its own module that takes in an object with details about the database engine, like you did here, makes sense to me.

It's not so much the complexity of the code, as the fact that I don't want code in my repositories that I'm not able to easily support.  It's just stress for me, and frustration for the people who run into bugs that I can't help them with.

**Pull request closed without merging**